src/accelerate/data_loader.py

```diff
@@ -364,10 +364,14 @@ def __iter__(self):
 
     @property
     def total_batch_size(self):
+        if self.batch_sampler is None:
+            batch_sampler = self.sampler
+        else:
+            batch_sampler = self.batch_sampler
```

Comment:
I think we should use a instance check on the sampler here, as in my tests, there was a `batch_sampler` when using a `BatchSampler` for `sampler`:
```py
batch_sampler = self.sampler if isinstance(self.sampler, BatchSampler) else self.batch_sampler
```

Reply:
Done 😄